### PR TITLE
add hg_pillar similar to git_pillar 

### PR DESCRIPTION
**Pull request by ptonelli**
originally from https://bitbucket.org/flub/hg_pillar

This pull requests also adds the first true unit test for custom ext pillars

### CODE REVIEW

File: salt/pillar/hg_pillar.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    '''
+    Return a list of branches that can be used as environments
+    '''
+
+
+def purge_cache():
+    '''
+    Purge the hg_pillar cache
+    '''
+
```
Comment (jfindlay) [inline salt/pillar/hg_pillar.py]: Are these functions intended to be implemented later, or are they just not needed at all?

**Comment by iggy** [comment]
No version annotations

**Comment by thatch45** [comment]
Thanks @ptonelli ! Lets get these things cleaned up so we can get this merged in!

File: salt/pillar/hg_pillar.py (hunk 1 above)
Comment (ptonelli) [inline salt/pillar/hg_pillar.py]: I believe they are for future implementation, but should be removed as they are not called in the hg_pillar.py. I am testing if things are working correctly without them

Comment (jfindlay) [inline salt/pillar/hg_pillar.py]: Great, thanks, @ptonelli.

**Comment by ptonelli** [comment]
@iggy on which version should I put the version annotations ?

**Comment by iggy** [comment]
I'd say Beryllium to start with... It seems like there's enough interest here that it shouldn't get pushed off a full release.

**Comment by jfindlay** [comment]
@ptonelli, @gracinet, excellent, thanks.

**Pull request merged**